src/adapter/mod.rs

```diff
@@ -187,6 +188,59 @@ impl Uuid {
     }
 }
 
+const UPPER: [u8; 16] = [
+    b'0', b'1', b'2', b'3', b'4', b'5', b'6', b'7', b'8', b'9', b'A', b'B',
+    b'C', b'D', b'E', b'F',
+];
+const LOWER: [u8; 16] = [
+    b'0', b'1', b'2', b'3', b'4', b'5', b'6', b'7', b'8', b'9', b'a', b'b',
+    b'c', b'd', b'e', b'f',
+];
+/// The segments of a UUID's [u8; 16] corresponding to each group.
+const BYTE_POSITIONS: [usize; 6] = [0, 4, 6, 8, 10, 16];
+/// The locations that hyphens are written into the buffer, after each
+/// group.
+const HYPHEN_POSITIONS: [usize; 4] = [8, 13, 18, 23];
+
+fn encode<'a>(
+    full_buffer: &'a mut [u8],
+    start: usize,
```

Comment:
Is there a reason to pass a start index in rather than expect `full_buffer` to have been sliced?

Reply:
Yeah, for the URN case: there's a prefix that gets added in front of what `encode` does, but I want to have that prefix included in the return value, and thus also in the `from_utf8_mut` call in this function.

Reply:
Ah gotcha :+1:

Reply:
I added a doc comment to `encode` that explains this.